lib/kubernetes-deploy/kubernetes_resource/persistent_volume_claim.rb

```diff
@@ -3,16 +3,76 @@ module KubernetesDeploy
   class PersistentVolumeClaim < KubernetesResource
     TIMEOUT = 5.minutes
 
+    def sync(cache)
+      super
+      @storage_classes = cache.get_all("StorageClass").map { |sc| StorageClass.new(sc) }
+    end
+
     def status
       exists? ? @instance_data["status"]["phase"] : "Not Found"
     end
 
     def deploy_succeeded?
-      status == "Bound"
+      return false if deploy_failed?
+      return true if status == "Bound"
+
+      # if the StorageClass has volumeBindingMode: WaitForFirstConsumer,
+      # it won't bind until after a Pod mounts it. But it must be pre-deployed,
+      # as the Pod requires it. So 'Pending' must be treated as a 'Success' state
+      if storage_class&.volume_binding_mode == "WaitForFirstConsumer"
+        status == "Pending" || status == "Bound"
```

Comment:
As far as kubernetes-deploy is concerned, this case is treated as `Success`. However, the critical point of actually binding the volume has still not occurred so it might be misleading to indicate success carte blanche. To that end, should we log a warning to the output?

Conversely, if the binding fails, the pod that's using the claim will fail anyway so 🤷‍♂

Reply:
I agree there is room to 🔫 👣 here. But I don't think its our job to warn people of the dangers of using `"WaitForFirstConsumer"`